presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardOrganizerUtil.java

```diff
@@ -72,4 +209,68 @@ static long determineDay(long rangeStart, long rangeEnd, Type type)
         long millisInEndDay = rangeEnd - Duration.ofDays(endDay).toMillis();
         return (millisInStartDay >= millisInEndDay) ? startDay : endDay;
     }
+
+    private static Optional<ShardRange> getShardRange(List<TableColumn> columns, ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<Object> minValuesBuilder = ImmutableList.builder();
+        ImmutableList.Builder<Object> maxValuesBuilder = ImmutableList.builder();
+
+        for (TableColumn tableColumn : columns) {
+            long columnId = tableColumn.getColumnId();
+            Type type = tableColumn.getDataType();
+
+            Object min = getValue(resultSet, type, minColumn(columnId));
+            Object max = getValue(resultSet, type, maxColumn(columnId));
+
+            minValuesBuilder.add(min);
+            maxValuesBuilder.add(max);
+        }
+
+        List<Type> types = columns.stream()
+                .map(TableColumn::getDataType)
+                .collect(toList());
+
+        Tuple minTuple = new Tuple(types, minValuesBuilder.build());
+        Tuple maxTuple = new Tuple(types, maxValuesBuilder.build());
+
+        return Optional.of(new ShardRange(minTuple, maxTuple));
+    }
+
+    private static String indexColumnNames(List<TableColumn> tableColumns)
+    {
+        StringJoiner joiner = new StringJoiner(", \n", ",\n", "").setEmptyValue("");
+
+        for (TableColumn tableColumn : tableColumns) {
+            joiner.add(minColumn(tableColumn.getColumnId()));
+            joiner.add(maxColumn(tableColumn.getColumnId()));
+        }
+        return joiner.toString();
+    }
+
+    private static Object getValue(ResultSet resultSet, Type type, String columnName)
+            throws SQLException
+    {
+        JDBCType jdbcType = jdbcType(type);
+        Object value = getValue(resultSet, type, columnName, jdbcType);
+        return resultSet.wasNull() ? null : value;
+    }
+
+    private static Object getValue(ResultSet resultSet, Type type, String columnName, JDBCType jdbcType)
+            throws SQLException
+    {
+        switch (jdbcType) {
+            case BOOLEAN:
+                return resultSet.getBoolean(columnName);
+            case INTEGER:
+                return resultSet.getInt(columnName);
+            case BIGINT:
+                return resultSet.getLong(columnName);
+            case DOUBLE:
+                return resultSet.getDouble(columnName);
+            case VARBINARY:
+                return wrappedBuffer(resultSet.getBytes(columnName)).toStringUtf8();
+        }
+        throw new PrestoException(RAPTOR_ERROR, "Unhandled type: " + type);
```

Comment:
This can be `IllegalArgumentException` since this code is not part of query execution